pkg/account/deployer/deployer.go

```diff
@@ -17,15 +17,288 @@
 package deployer
 
 import (
+	"context"
+	"fmt"
+	accountmanagement "github.com/dynatrace/dynatrace-configuration-as-code-core/gen/account_management"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/account"
-	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/manifest"
+)
+
+type (
+	AccountInfo struct {
+		Name        string
+		AccountUUID string
+	}
+	localId     = string // local (monaco related) identifier
+	envName     = string // dt environment name
+	remoteId    = string // dt entity identifier
+	projectName = string // monaco project name
+	idLookupFn  func(id localId) remoteId
 )
 
 type Options struct {
 	DryRun bool
 }
 
-// Deploy deploys all given resources to all given accounts
-func Deploy(accounts map[string]manifest.Account, resources map[string]*account.Resources, opts Options) error {
+//go:generate mockgen -source=deployer.go -destination=client_mock.go -package=deployer client
+type client interface {
+	getGlobalPolicies(ctx context.Context) (map[string]remoteId, error)
+	getManagementZones(ctx context.Context) ([]ManagementZone, error)
+	upsertPolicy(ctx context.Context, policyLevel string, policyLevelId string, policyId string, policy Policy) (remoteId, error)
+	upsertGroup(ctx context.Context, groupId string, group Group) (remoteId, error)
+	upsertUser(ctx context.Context, userId string) (remoteId, error)
+	updateAccountPolicyBindings(ctx context.Context, groupId string, policyIds []string) error
+	updateEnvironmentPolicyBindings(ctx context.Context, envName string, groupId string, policyIds []string) error
+	updateGroupBindings(ctx context.Context, userId string, groupIds []string) error
+	updatePermissions(ctx context.Context, groupId string, permissions []accountmanagement.PermissionsDto) error
+	getAccountInfo() AccountInfo
+}
+
+type AccountDeployer struct {
+	accountManagementClient client
+
+	deployedPolicies       map[localId]remoteId
+	deployedGroups         map[localId]remoteId
+	deployedMgmtZones      []accountmanagement.ManagementZoneResourceDto
+	availablePermissionIds []remoteId
+}
+
+func NewAccountDeployer(client client) *AccountDeployer {
+	return &AccountDeployer{
+		accountManagementClient: client,
+
+		deployedPolicies:       make(map[localId]remoteId),
+		deployedGroups:         make(map[localId]remoteId),
+		availablePermissionIds: make([]remoteId, 0),
+	}
+}
+
+func (d *AccountDeployer) Deploy(resources map[projectName]*account.Resources, opts Options) error {
+	if opts.DryRun {
+		log.Info("Dry run not yet implemented... nothing to do...")
+		return nil
+	}
+
+	var err error
+	d.deployedPolicies, err = d.accountManagementClient.getGlobalPolicies(context.TODO())
+	if err != nil {
+		return err
+	}
+
+	d.deployedMgmtZones, err = d.accountManagementClient.getManagementZones(context.TODO())
+	if err != nil {
+		return err
+	}
+
+	for _, res := range resources {
```

Comment:
Looking at this usage: Does it actually have any benefit to have the project -> resources mapping here?
If I'm not missing anything you're just looping over the resources and never even use the project name for anything. 

If you'd build a single combined account.Resources object when loading the projects, this deployer could simply deploy all policies, groups and users it got. 🤔

Also in that case we could flag user errors, like defining the same group twice in different projects, which I think we're missing right now.

Reply:
lets think about that in another PR?

Reply:
fine by me 👍 

As discussed directly, it might also make sense to just modify the persistence/account/loader to target several folders instead of a single one, as it already does this level of validation and builds a single struct for everything it finds.